server/src/test/java/org/opensearch/index/engine/InternalEngineTests.java

```diff
@@ -342,6 +342,61 @@ public void testVerboseSegments() throws Exception {
         }
     }
 
+    public void testSegmentsWithUseCompoundFileFlag_true() throws IOException {
+        try (Store store = createStore(); Engine engine = createEngine(defaultSettings, store, createTempDir(), new TieredMergePolicy())) {
+            ParsedDocument doc = testParsedDocument("1", null, testDocument(), B_1, null);
+            Engine.Index index = indexForDoc(doc);
+            engine.index(index);
+            engine.flush();
+            final List<Segment> segments = engine.segments(false);
+            assertThat(segments.size(), equalTo(1));
```

Comment:
Nitpick, but you can use `assertThat(segments, hasSize(1))` which will generate a slightly better error message on failure.